torch/csrc/jit/graph_executor.cpp

```diff
@@ -418,6 +419,9 @@ struct GraphExecutorImpl {
     // decisions to insert/remove undefs nodes and to work before
     // we propagate input shapes.
 
+    // Decompose addmm nodes to add + mm, so expands can be inserted and
+    // gradients accumulated on the backward pass
+    DecomposeAddmm(g);
```

Comment:
In the future, we should have one "Peephole" pass that does pre-shape analysis simplifications like this. For now while this is only addmm, we can keep it as is, but it would be useful to have a comment saying what should happen if we find additional nodes that need canonicalization.